services/src/main/java/org/keycloak/protocol/oidc/AccessTokenIntrospectionProvider.java

```diff
@@ -97,6 +117,53 @@ public Response introspect(String token) {
         }
     }
 
+    private AccessToken getAccessToken(String token) throws VerificationException {
+        TokenVerifier<AccessToken> verifier = TokenVerifier.create(token, AccessToken.class);
+        SignatureVerifierContext verifierContext = session.getProvider(SignatureProvider.class, verifier.getHeader().getAlgorithm().name()).verifier(verifier.getHeader().getKeyId());
+        verifier.verifierContext(verifierContext);
```

Comment:
This is the verifyAccessToken method except the return statement. 
I propose create a new method used here and in verifyAccessToken method.
If verification failed now Keycloak return failed. With your code an exception will be thrown. 
I believe that we should keep same behaviour and return here null.